mesonbuild/build.py

```diff
@@ -1100,7 +1101,7 @@ def determine_filenames(self, is_cross, env):
         # C, C++, Swift, Vala
         # Only Windows uses a separate import library for linking
         # For all other targets/platforms import_filename stays None
-        elif for_windows(is_cross, env):
+        elif for_windows(is_cross, env) or for_cygwin(is_cross, env):
```

Comment:
Probably better to split this off into its own `elif` conditional since you can't build for Cygwin with MSVC.